sdk/cmake/host-tools.cmake

```diff
@@ -26,6 +26,48 @@ function(setup_host_tools)
         set(HOST_MODULE_SUFFIX ".so")
     endif()
 
+    # Normalize to the same format as our own ARCH, and add one for the VC shell
+    string(TOLOWER "${CMAKE_HOST_SYSTEM_PROCESSOR}" lowercase_CMAKE_HOST_SYSTEM_PROCESSOR)
+    if(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86 OR lowercase_CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "^i[3456]86$")
+        set(HOST_ARCH i386)
+        set(VCVARSALL_ARCH x86)
+    elseif(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86_64 OR lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL amd64)
+        set(HOST_ARCH amd64)
+        set(VCVARSALL_ARCH x64_x86) # host-tools are not happy compiling natively for x64...
+    elseif(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL arm)
+        set(HOST_ARCH arm)
+        set(VCVARSALL_ARCH arm)
+    else()
+        message(FATAL_ERROR "Unknown host architecture: ${lowercase_CMAKE_HOST_SYSTEM_PROCESSOR}")
+    endif()
+
+    if(ARCH STREQUAL HOST_ARCH)
```

Comment:
This could as well be `if ((NOT MSVC) OR (CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL WINARCH))`

And put the not-nice part of setting VCVARSALL related in the no-true case

Reply:
This is more explicit, I prefer it that way.